accounts/abi/bind/backends/remote.go

```diff
@@ -0,0 +1,202 @@
+	Method  string        `json:"method"`  // Remote procedure name to invoke on the server
+	Params  []interface{} `json:"params"`  // List of parameters to pass through (keep types simple)
+}
+
+// response is a JSON RPC response package sent back from the API server.
+type response struct {
+	JsonRpc string          `json:"jsonrpc"` // Version of the JSON RPC protocol, always set to 2.0
+	Id      int             `json:"id"`      // Auto incrementing ID number for this request
+	Error   json.RawMessage `json:"error"`   // Any error returned by the remote side
+	Result  json.RawMessage `json:"result"`  // Whatever the remote side sends us in reply
+}
+
+// request forwards an API request to the RPC server, and parses the response.
+//
+// This is currently painfully non-concurrent, but it will have to do until we
+// find the time for niceties like this :P
+func (backend *rpcBackend) request(method string, params []interface{}) (json.RawMessage, error) {
+	backend.lock.Lock()
+	defer backend.lock.Unlock()
+
+	// Ugly hack to serialize an empty list properly
+	if params == nil {
+		params = []interface{}{}
+	}
+	// Assemble the request object
+	req := &request{
+		JsonRpc: "2.0",
+		Id:      int(atomic.AddUint32(&backend.autoid, 1)),
+		Method:  method,
+		Params:  params,
+	}
+	if err := backend.client.Send(req); err != nil {
+		return nil, err
+	}
+	res := new(response)
+	if err := backend.client.Recv(res); err != nil {
+		return nil, err
+	}
+	if len(res.Error) > 0 {
+		return nil, fmt.Errorf("remote error: %s", string(res.Error))
+	}
+	return res.Result, nil
+}
+
+// ContractCall implements ContractCaller.ContractCall, delegating the execution of
+// a contract call to the remote node, returning the reply to for local processing.
+func (b *rpcBackend) ContractCall(contract common.Address, data []byte, pending bool) ([]byte, error) {
+	// Pack up the request into an RPC argument
+	args := struct {
+		To   common.Address `json:"to"`
+		Data string         `json:"data"`
+	}{
+		To:   contract,
+		Data: common.ToHex(data),
+	}
+	// Execute the RPC call and retrieve the response
+	block := "latest"
+	if pending {
+		block = "pending"
+	}
+	res, err := b.request("eth_call", []interface{}{args, block})
+	if err != nil {
+		return nil, err
+	}
+	var hex string
+	if err := json.Unmarshal(res, &hex); err != nil {
+		return nil, err
+	}
+	// Convert the response back to a Go byte slice and return
+	return common.FromHex(hex), nil
+}
+
+// AccountNonce implements ContractTransactor.AccountNonce, delegating the
+// current account nonce retrieval to the remote node.
+func (b *rpcBackend) AccountNonce(account common.Address) (uint64, error) {
+	res, err := b.request("eth_getTransactionCount", []interface{}{account.Hex(), "pending"})
+	if err != nil {
+		return 0, err
+	}
+	var hex string
+	if err := json.Unmarshal(res, &hex); err != nil {
+		return 0, err
+	}
+	return new(big.Int).SetBytes(common.FromHex(hex)).Uint64(), nil
```

Comment:
Actually, neither `SetString` or `common.String2Big` (which uses `SetString` internally) is appropriate, as the result of the RPC call is prefixed with `0x`, which `SetString` cannot parse: http://play.golang.org/p/lqdQPiPCiP .